nighthawk/test/client_test.cc

```diff
@@ -0,0 +1,147 @@
+#include <chrono>
+
+#include "gtest/gtest.h"
+
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/stats/mocks.h"
+
+#include "test/integration/http_integration.h"
+#include "test/integration/integration.h"
+#include "test/integration/utility.h"
+#include "test/server/utility.h"
+#include "test/test_common/utility.h"
+
+#include "common/api/api_impl.h"
+#include "common/common/thread_impl.h"
+#include "common/filesystem/filesystem_impl.h"
+
+#include "nighthawk/test/mocks.h"
+
+#include "nighthawk/source/client/client.h"
+#include "nighthawk/source/client/factories_impl.h"
+#include "nighthawk/source/client/options_impl.h"
+
+using namespace std::chrono_literals;
+
+namespace Nighthawk {
+namespace Client {
+
+class ClientTest : public Envoy::BaseIntegrationTest,
+                   public testing::TestWithParam<Envoy::Network::Address::IpVersion> {
+public:
+  ClientTest() : Envoy::BaseIntegrationTest(GetParam(), realTime(), ClientTest::envoy_config) {}
+
+  static void SetUpTestCase() {
+    Envoy::Filesystem::InstanceImplPosix file_system;
+    envoy_config = file_system.fileReadToEnd(Envoy::TestEnvironment::runfilesPath(
+        "nighthawk/test/test_data/benchmark_http_client_test_envoy.yaml"));
+    envoy_config = Envoy::TestEnvironment::substitute(envoy_config);
+  }
+
+  void SetUp() override {
+    // We fork the integration test fixture into a child process, to avoid conflicting
+    // runtimeloaders as both NH and the integration server want to own that and we can have only
+    // one. The plan is to move to python for this type of testing, so hopefully we can deprecate
+    // this test and it's peculiar setup with fork/pipe soon.
+    pipe(fd_port);
```

Comment:
Can you wrap every single call, `pipe`, `write`, `read`, etc. with the expected return and a `RELEASE_ASSERT`? This would make me a lot more comfortable with the robustness here.